### PR TITLE
[Admin API] [Order] Remove locale from require data during cart creation

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets |  |
| License         | MIT |

A customer can change the language during checkout, so we should also allow for this in AdminAPI.

### CODE REVIEW

File: src/Sylius/Bundle/AdminApiBundle/Form/Type/CompleteType.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use Sylius\Bundle\ResourceBundle\Form\DataTransformer\ResourceToIdentifierTransformer;
+use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\ReversedTransformer;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
```
Comment (tuka217) [inline src/Sylius/Bundle/AdminApiBundle/Form/Type/CompleteType.php]: I think, that you are the author of this class (also or only?)

File: docs/api/checkouts.rst [hunk 2]
```diff
@@ -957,3 +957,33 @@ Exemplary Response
 .. code-block:: text
 
     STATUS: 204 No Content
+
+Last possible action during order completion is changing the order locale:
```
Comment (NeverResponse) [inline docs/api/checkouts.rst]: Quite sure this is a cart modification, not order's, besides it gets a bit confusing when there are **2** last final actions (notes/locale).
Also they're both going through with `PATCH`, seems both of them could be merged as one example with **every** final change to the cart, before it becomes an order.

File: docs/api/carts.rst [hunk 3]
```diff
@@ -210,6 +207,7 @@ Exemplary Response
 .. note::
 
     A currency code will be added automatically based on the channel settings. Read more about channels :doc:`here </book/configuration/channels>`.
+    Same rule applies locale codes. Locale can be changed during complete request when finalizing a cart.
```
Comment (NeverResponse) [inline docs/api/carts.rst]: `(...) to locale codes.`,
The last sentence sounds a whole lot like a pleonasm. Better phrasing would be much appreciated.

File: UPGRADE.md [hunk 4]
```diff
@@ -96,6 +96,9 @@ You can bring back previous configuration by overriding current routing with you
 
 * Bundle was renamed from `ApiBundle` to `AdminApiBundle`. Routing and config was changed from `sylius_api` to `sylius_admin_api`.
 
+* Don't send `localeCode` during cart creation. If default channel locale is not enough (it is assigned dynamically), send it with complete request.
+  Also, complete request should not be send with `PUT` method by default, as it will set null value on locale code, which will result in invalid argument exception.
```
Comment (NeverResponse) [inline UPGRADE.md]: `by default` is not necessary

File: docs/api/checkouts.rst [hunk 5]
```diff
@@ -899,7 +899,7 @@ Definition
 
 .. code-block:: text
 
-    PUT /api/v1/checkouts/complete/{id}
+    PUT/PATCH /api/v1/checkouts/complete/{id}
```
Comment (tuka217) [inline docs/api/checkouts.rst]: So, when we should use `PUT`?

**Pull request closed without merging**